### ISSUE TITLE
Provide a reliable way of setting camera to current coordinate

### ISSUE THREAD

**Issue by azarovalex** [mapbox/mapbox-navigation-ios-examples#101]
Currently in many samples we have TODOs like here:

https://github.com/mapbox/mapbox-navigation-ios-examples/blob/42094c44ec8c322201fc2c3e221d89b9be76284c/Navigation-Examples/Examples/Advanced.swift#L48 https://github.com/mapbox/mapbox-navigation-ios-examples/blob/42094c44ec8c322201fc2c3e221d89b9be76284c/Navigation-Examples/Examples/Building-Extrusion.swift#L59

Example code shouldn’t have to-do items, so we need to find a way to use new camera API and improve existing code.

**Comment by azarovalex** [comment]
This TODO block code
```
DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
            if let coordinate = self.navigationMapView.mapView.location.latestLocation?.coordinate {
                let cameraOptions = CameraOptions(center: coordinate, zoom: 13.0)
                self.navigationMapView.mapView.camera.setCamera(to: cameraOptions)
            }
        }
```
 can be replaced with 
 ```navigationMapView.navigationCamera.viewportDataSource = NavigationViewportDataSource(navigationMapView.mapView, viewportDataSourceType: .raw)```
 
This will automatically zoom to the user location after the map view appears.
 
 Currently, Maxim replace removes this TODO blocks in https://github.com/mapbox/mapbox-navigation-ios-examples/pull/92
 When the PR lands, we need to verify that all todos were removed and the camera zooms to the current coordinate as before.

**Comment by azarovalex** [comment]
Currently waiting for https://github.com/mapbox/mapbox-navigation-ios-examples/pull/92 to land.

**Comment by 1ec5** [comment]
Will be fixed as part of mapbox/mapbox-navigation-ios#2946.

**Comment by 1ec5** [comment]
Fixed in #92.
